xtext/editorserver/src/websockets.js

```diff
@@ -59,15 +60,16 @@ function get_response(){
 const wss = new WebSocketServer({ port: 8000 });
 
 wss.on('connection', function connection(ws) {
-    var buildPathWatcher = deployPathWatcher = null;
+    const buildPathWatcher = null;
+    const deployPathWatcher = null;
     ws.isAlive = true;
     ws.on('error', () => {
         buildPathWatcher?.close();
         deployPathWatcher?.close();
     });
     ws.on('pong', () => { ws.isAlive = true; });
     ws.on('message', function(data) {
-        subscribe_to_build(data, buildPathWatcher, deployPathWatcher);
+        subscribe_to_build(this, data, buildPathWatcher, deployPathWatcher);
```

Comment:
What's this `this`? Why not use `ws` for clarity?

```suggestion
        subscribe_to_build(ws, data, buildPathWatcher, deployPathWatcher);
```

Reply:
Because we are calling this within a callback and this `refers` to `ws`.

Reply:
Sure, but that means as a reader of your code I have to do a lot more mental arithmetics than if you simply used `ws`...

Reply:
Changing the scope of file handles, this goes away.

Reply:
Not sure I understand :-)

Reply:
That's how JavaScript callbacks work!

Reply:
Sorry, if it's cryptic to read it means it's more difficult to maintain and easier to get wrong for future developers. Code is written for humans and readability should come first, surely.

Reply:
True, but a JavaScript developer should know how to deal with callback execution contexts.

Reply:
Sorry, I think I will have to insist on this. Assuming that `this` will point to `ws` is making an assumption on how the event handler function will actually be called. To confirm this, a reader of the code would have to dig deep into the documentation of the ws library (in fact, I couldn't confirm this assumption so far). In fact, typical example code for this uses `ws` explicitly. It's clearer, doesn't create any meaningful overhead -- there really is no reason not to do it.

Reply:
Sorry but I find it hard to understand as this causes redundancy. One has to read about the web sockets library anyway when working on this code whether it's using `this` or `ws`. 
I switched to an arrow function and passed `ws` as a parameter despite thinking it's redundant because I believe this is taking far too long as a discussion.